_posts/announcing-wcgi.md

```diff
@@ -0,0 +1,241 @@
+---
+title: "Announcing WCGI - Revolutionizing Server Development with WebAssembly"
+ogImage:
+    url: "TODO"
+excerpt: "TODO"
+date: "2023-04-04T12:00:00.000Z"
+coverImage: "TODO"
+author: Michael Bryan
+published: false
+---
+
+Welcome to the future of server-side development with WebAssembly! Introducing
+WCGI, a groundbreaking technology that marries the power of WebAssembly with the
+versatility and simplicity of CGI, paving the way for a new era in serverless
+applications.
+
+WCGI is built on a straightforward idea: take an existing CGI program and
+compile it to WebAssembly. The real innovation, however, lies in how `wasmer`
+runs that WebAssembly.
+
+Here are some of WCGI's highlights:
+
+- Benefit from decades of application and library development without learning
+  another web framework
+- Simplify distribution and release management by publishing your package on
+  WAPM
+- Enjoy faster cold start times as WebAssembly module instantiation is much
+  quicker than other serverless technologies
+- Reduce release artifact size – WAPM packages only contain your business logic
+  and static assets, eliminating the need for an HTTP stack or bulky Docker
+  containers
+- Use any language compatible with WASI, such as AssemblyScript, C, C++, Go,
+  JavaScript, PHP, Python, Ruby, Rust, or Swift
+- Experience enhanced security – WebAssembly code runs in a sandbox, with one
+  instance per request
+
+
+Picture a Wordpress installation without worrying about attackers breaking into
+your system.
+
+With WCGI, it's a reality.
+
+(video of Wasmer running WordPress)
+
+## The Why Behind WCGI
+
+When venturing into serverless solutions at Wasmer, we faced a crucial question:
+should we create our own framework and risk locking developers into a walled
+garden, or should we adopt an open standard that allows them to utilize existing
+code?
+
+CGI's alignment with the goal of executing a program per HTTP request makes it a
+compelling choice. Interestingly, CGI can outperform many other serverless
+solutions (such as WSGI in Python/Ruby or NodeJS) in terms of scalability and
+latency.
+
+Furthermore, it also closely mirrors the workers proposal from the **[Winter
+Community
+Group](https://wintercg.org/)**.
+
+Consider the challenge of running PHP programs on servers. We have two primary
+options:
+
+1. Wrap the PHP interpreter with a custom layer that instruments each HTTP call.
+2. Use the existing `php-cgi` program and simply compile it to Wasm.
+
+Option 2 is not only faster, but it also enables any web application on Wasmer
+more efficiently.
+
+By embracing WCGI, those seeking to achieve greater efficiency, security, and flexibility in server-side development can truly benefit from this innovative approach.
+
+## Creating a WCGI Application with Rust
+
+To create a WCGI application using Rust, first add the **`cgi`** crate as a
+dependency in your **`Cargo.toml`** file:
+
+```bash
```

Comment:
All the `bash` entries need to be modified to `shell-session` as bash modifier doesn't work well with our current markdown implementation.